content/posts/gitops/index.md

```diff
@@ -0,0 +1,497 @@
+
+The `prd` environment of our application is defined in its own `Application` object, where only a few parameters differ:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: prd
+  # ...
+spec:
+  source:
+    repoURL: https://github.com/bespinian/argocd-demo.git
+    # ...
+    path: env/prd
+  destination:
+    # ...
+    namespace: awesome-gitops-prd
+  syncPolicy:
+    # ...
+```
+
+Obviously, the `name` of our the productive `Application` is different. The Git repo is the same but the `path` points to the Kustomization for the productive environment in this case. Furthermore, the destination namespace is the productive one in this case.
+
+If we check the Kustomizations of our two environments we see that the `prd` environment is currently running version `1.0.0` of our application, whereas the `int` environment is already on version `2.0.0`.
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
+  - ../../resources
+images:
+  - name: bespinian/awesome-image
+    newTag: 1.0.0
+configMapGenerator:
+  - name: awesome-config
+    literals:
+      - appTitle=Something awesome!
+      - dbHost=db.bespinian.io
+      - appVersion=1.0.0
+```
+
+Indeed, if we have a look in [Docker Hub](https://registry.hub.docker.com/r/bespinian/awesome-image/tags), which is the artifact store we use for our release process in this example, we can see that there are versions `1.0.0` and `2.0.0` of the `awesome-image` which have been released by our build process.
+
+![](./argocd-dockerhub.svg)
+
+So let's consider how we upgrade our `prd` environment to version `2.0.0` using Gitops. In order to achieve this, we simply edit our declarative representation, which in this case is our Kustomization. Let's assume we come up with the following new version.
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
+  - ../../resources
+images:
+  - name: bespinian/awesome-image
+    newTag: 2.0.0
+configMapGenerator:
+  - name: awesome-config
+    literals:
+      - appTitle=Something even more awesome!
+      - dbHost=db.bespinian.io
+      - appVersion=2.0.0
+```
+
+Our edits will lead to the following Git diff
+
+```shell
```

Comment:
I think, there is a proper `diff` code type that would look much nicer, but we have to check if it's supported by our Hugo theme. If not, `txt` would make more sense though because `shell` is for shell commands and their outputs.

Reply:
Done.